telegram/_files/inputfile.py

```diff
@@ -35,7 +34,11 @@ class InputFile:
     """This object represents a Telegram InputFile.
 
     .. versionchanged:: 20.0
-        The former attribute ``attach`` was renamed to :attr:`attach_name`.
+
+        * The former attribute ``attach`` was renamed to :attr:`attach_name`.
+        * Method ``is_image`` was removed. If you pass :obj:`bytes` to :paramref:`obj` and would
```

Comment:
If I understand the python [docs](https://docs.python.org/3.12/library/mimetypes.html#mimetypes.guess_type) correctly, we can just pass `guess_type(obj)` since it says it supports a path like object from python 3.8 onwards? That perhaps increases accuracy?

Reply:
Don't think that this would work. [path-like](https://docs.python.org/3.12/glossary.html#term-path-like-object) != [file-like](https://docs.python.org/3/glossary.html?highlight=glossary#term-file-object)

Reply:
I just tested passing a `Path` object to `guess_type` and it works. Probably cause a `path` like object extends from `PurePath` like it said in the [docs](https://docs.python.org/3.12/library/os.html#os.PathLike)

Reply:
ah but we have it open()ed already, so this won't work hmm

Reply:
my gut feeling is that `InputFile` should represent the raw data to be uploaded 🤔 We could try and accept path objects for the bot methods/`parse_file_input` though, but that's out of scope for this PR IMO. See also https://github.com/python-telegram-bot/python-telegram-bot/projects/7#card-79049723